src/line/funcs/testAddValues.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @jest-environment jsdom
+ */
+const fs = require('fs');
+require('@testing-library/jest-dom');
+const domTesting = require('@testing-library/dom');
+const userEvent = require('@testing-library/user-event').default;
+const functions = require('./functions.js');
+
+const [htmlPath, jsPath] = functions.getPaths();
+
+// this function will perform the first test given in the "UI integration 
+// tests" section of the assignment document. The specific test is labeled as
+// "adding values in the chart builder".
+async function testAddValues(elements, inputs, num_empty_inputs){
+
+	// do initializations: load in files that we're gonna be testing, init
+	// expected length of (X, Y) pairs, grab the input fields, the '+' 
+	// button, and init our user for events
+    functions.initDomFromFiles(htmlPath, jsPath);
+	functions.initElements(elements);
+	const user = userEvent.setup();
+	var expected_length = 1;
+
+	// do assertion: quick cehck that we only have 1 (X, Y) input available to 
+	// us to start off.
+	functions.assertInputLength(elements, expected_length);
+
+	for(const input of inputs){
+
+		// do actions: create new (X, Y) inputs and add the above values 
+		// (stored in input) to them
+		await user.type(elements.X_inputs[expected_length - 1], input[0]);
+		await user.type(elements.Y_inputs[expected_length - 1], input[1]);
+		await user.click(elements.add_input_button);
+		expected_length++;
+
+		// do some assertions: make sure that our (X, Y) inputs increment
+		// with each press of the add_input_button
+		functions.assertInputLength(elements, expected_length);
+
+	}
+
```

Comment:
Ideally, the section of code above, the for loop, and the code below should be separate functions. Adding values, and adding empty boxes should not be tested together as one behavior, but as two separate behaviors (that could work together, as another separate test).

Reply:
Made separate tests for next PR. first checks we can add empty input boxes. second checks we can add empty input boxes and fill them with values. third checks we can add inputs, fill those inputs, and further input box creation will not affect previous boxes input values.